File: JavaScript/Math-Array-String/54.js
// 1688. Count of Matches in Tournament


/**
 * @param {number} n
 * @return {number}
 */
var numberOfMatches = function(n) {
    let totalMatches = 0;
    while (n > 1) {
        if (n % 2 === 0) {
            totalMatches += n / 2;
            n = n / 2;
        } else {
            totalMatches += (n - 1) / 2;
            n = (n - 1) / 2 + 1;
        }
    }
    return totalMatches;
};

console.log(numberOfMatches(7));  // Output: 6
console.log(numberOfMatches(14)); // Output: 13

File: Python/Math-Array-String/54.py
# 1688. Count of Matches in Tournament

class Solution:
    def numberOfMatches(self, n: int) -> int:
        total_matches = 0
        while n > 1:
            if n % 2 == 0:
                matches = n // 2
                n = n // 2
            else:
                matches = (n - 1) // 2
                n = (n - 1) // 2 + 1
            total_matches += matches
        return total_matches
    
sol = Solution()
output = sol.numberOfMatches(7)
print(output) # Output: 6